lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1057,6 +1058,47 @@ def remove_certificate(self, certificate: str) -> None:
         for certificate_relation in certificates_relation:
             self._remove_certificate(certificate=certificate, relation_id=certificate_relation.id)
 
+    def get_all_relations_certificates(self) -> Dict[str, Dict[str, str]]:
+        """Returns a dictionary of all issued certificates.
+
+        Certificates are returned per application name and CSR.
+
+        Returns:
+            dict: Certificates per application name.
+        """
+        certificates: Dict[str, Dict[str, str]] = defaultdict(dict)
+        for relation in self.model.relations[self.relationship_name]:
+            provider_relation_data = _load_relation_data(relation.data[self.charm.app])
+            provider_certificates = provider_relation_data.get("certificates", [])
+            for certificate in provider_certificates:
+                certificates[relation.app.name].update(  # type: ignore[union-attr]
+                    {certificate["certificate_signing_request"]: certificate["certificate"]}
+                )
+        return certificates
+
+    def get_relation_certificates(self, relation_id: int) -> Dict[str, str]:
+        """Returns a list of all issued certificates for a given relation.
```

Comment:
Docstring doesn't match type hint.
But I wonder if we should have just one method for both cases.